browser/gui/app.cpp

```diff
@@ -123,6 +123,8 @@ App::App(std::string browser_title, std::string start_page_hint, bool load_start
         spdlog::critical("imgui-sfml initialization failed");
         std::abort();
     }
+    // ImGui tries to write to the current working directory. This is problematic for a number of reasons.
+    ImGui::GetIO().IniFilename = std::getenv("HASTUR_STATE_FILENAME");
```

Comment:
I've thought a bit more about it, and I don't think this is how we want to do this. Not saving anything by default isn't very reasonable, and this env var won't work once we start e.g. caching things and saving browser data outside of whatever imgui is doing right now.

How about either making this respect XDG, and/or adding an env var or flag that tells the browser to never touch the disk? Both seem like reasonable options and things that we'd want in the future as well.

Reply:
I have a few worries about imgui's IO design. I'm assuming it's gonna be short term thing, I don't want to allocate to many cycles for it if it's gonna be temporary. I'm happy to make conf write an opt-out envvar if that's better? But I don't want to have to teach it about XDG?

Reply:
It being an disk IO opt-out env var is fine, but we'll probably use the same env var to block all disk IO, so give it a good name. :P

XDG was just a thought and something someone else can add if they ever feel like it. We'll need to teach hastur a bit about XDG in the future for when we actually need to store browser settings and caches and things either way.

Reply:
> XDG was just a thought and something someone else can add if they ever feel like it. We'll need to teach hastur a bit about XDG in the future for when we actually need to store browser settings and caches and things either way.

Agreed, I actually considered if now was the right time to teach hastur about XDG, but the end decision was "soon" over now.